utils/data-store/index.ts

```diff
@@ -0,0 +1,77 @@
+import { HttpServerError } from '#clients/exceptions';
+import { FetchRessourceException } from '#models/exceptions';
+import { logWarningInSentry } from '#utils/sentry';
+
+/**
+ * Caches a promise response in memory
+ *
+ * When gave a TTR, it will attempt to refresh data every TTR milliseconds.
+ * As long as it as a data in store it will ignore any failed attempt to refresh and still return outdated cached data
+ */
+export class DataStore<T> {
```

Comment:
Ce n'est pas un singleton : il peut y avoir plusieurs instances de DataStore qui coexistent.

Néanmoins chaque type d'instance (storeName) du DataStore est actuellement instantié une seule fois du fait de leur encapsulation au sein de variables globales communes à l'ensemble des requêtes (superAgentsList, clientIdccMetadata, etc).

Reply:
Yes en effet. Après par singleton, je faisais plutot référence a l'utilisation de `this.onGoingRefresh` pour dire que la requête de refresh doit forcément être unique

Reply:
Mais je suis d'accord, le terme est abusif